worker/import.go

```diff
@@ -0,0 +1,303 @@
+/*
+ * SPDX-FileCopyrightText: © Hypermode Inc. <hello@hypermode.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package worker
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	apiv25 "github.com/dgraph-io/dgo/v250/protos/api.v25"
+	"github.com/dgraph-io/ristretto/v2/z"
+	"github.com/dustin/go-humanize"
+	"github.com/golang/glog"
+	"github.com/hypermodeinc/dgraph/v25/conn"
+	"github.com/hypermodeinc/dgraph/v25/posting"
+
+	"github.com/hypermodeinc/dgraph/v25/protos/pb"
+
+	"github.com/hypermodeinc/dgraph/v25/schema"
+	"github.com/pkg/errors"
+)
+
+func ProposeDrain(ctx context.Context, drainMode *pb.Drainmode) error {
+	memState := GetMembershipState()
```

Comment:
instead of this, can we do similar to what we are doing in restore:
```
if groups().ServesGroup(req.GetGroupId()) && groups().Node.AmLeader() {
		_, err := (&grpcWorker{}).Restore(ctx, req)
		return err
	}

	pl := groups().Leader(req.GetGroupId())
	if pl == nil {
		return conn.ErrNoConnection
	}
	con := pl.Get()
	c := pb.NewWorkerClient(con)

	_, err := c.Restore(ctx, req)
	return err
```

Reply:
This setup is solely for enabling drain mode; no streaming is involved at this stage. What you're suggesting might be applicable when implementing streaming in an HA + sharded cluster configuration.